components/scream/src/share/atm_process/atmosphere_diagnostic.cpp

```diff
@@ -0,0 +1,28 @@
+#include "share/atm_process/atmosphere_diagnostic.hpp"
+
+namespace scream
+{
+
+AtmosphereDiagnostic::
+AtmosphereDiagnostic (const ekat::Comm& comm, const ekat::ParameterList& params)
+  : AtmosphereProcess(comm,params)
+{
+  // Nothing to do here
+}
+
+// Function to retrieve the diagnostic output which is stored in m_diagnostic_output
+Field& AtmosphereDiagnostic::get_diagnostic (const Real dt) {
+  run(dt);
+  return m_diagnostic_output;
```

Comment:
Safety guard:
```
return m_diagnostic_output.get_const();
```
which forces customers to only extract const data from the field (cannot be written on*).


\* Well, they can still take the const data out, cast away const, overwrite the data, then do an evil laugh, but what you gonna do?